### PR TITLE
Bugfix jaeger exporter test panic

### LINKED ISSUE #1870: Jaeger exporter test failure: panic: close of closed channel
### Description

Running the test suite resulted in a panic for the Jaeger exporter tests.

### Environment

- OS: Mac OS X 10.15.7  19H524
- Architecture: amd64
- Go Version: 1.15
- opentelemetry-go version: v0.20.0

### Steps To Reproduce

```
go test -timeout 60s -short ./...
ok  	go.opentelemetry.io/otel	0.170s
ok  	go.opentelemetry.io/otel/attribute	0.180s
ok  	go.opentelemetry.io/otel/baggage	0.153s
ok  	go.opentelemetry.io/otel/codes	0.143s
?   	go.opentelemetry.io/otel/internal	[no test files]
ok  	go.opentelemetry.io/otel/internal/baggage	0.181s
ok  	go.opentelemetry.io/otel/internal/global	0.395s
ok  	go.opentelemetry.io/otel/internal/internaltest	0.164s
?   	go.opentelemetry.io/otel/internal/matchers	[no test files]
?   	go.opentelemetry.io/otel/internal/metric	[no test files]
?   	go.opentelemetry.io/otel/internal/trace/noop	[no test files]
ok  	go.opentelemetry.io/otel/propagation	0.189s
ok  	go.opentelemetry.io/otel/semconv	0.243s
?   	go.opentelemetry.io/otel/unit	[no test files]
go test -timeout 60s -short ./bridge/opencensus/...
ok  	go.opentelemetry.io/otel/bridge/opencensus	0.154s
ok  	go.opentelemetry.io/otel/bridge/opencensus/utils	0.165s
go test -timeout 60s -short ./bridge/opentracing/...
ok  	go.opentelemetry.io/otel/bridge/opentracing	0.155s
?   	go.opentelemetry.io/otel/bridge/opentracing/internal	[no test files]
?   	go.opentelemetry.io/otel/bridge/opentracing/migration	[no test files]
go test -timeout 60s -short ./exporters/metric/prometheus/...
ok  	go.opentelemetry.io/otel/exporters/metric/prometheus	0.226s
go test -timeout 60s -short ./exporters/otlp/...
ok  	go.opentelemetry.io/otel/exporters/otlp	1.171s
ok  	go.opentelemetry.io/otel/exporters/otlp/internal/otlpconfig	0.210s
?   	go.opentelemetry.io/otel/exporters/otlp/internal/otlptest	[no test files]
ok  	go.opentelemetry.io/otel/exporters/otlp/internal/transform	0.203s
ok  	go.opentelemetry.io/otel/exporters/otlp/otlpgrpc	10.557s
ok  	go.opentelemetry.io/otel/exporters/otlp/otlphttp	15.546s
go test -timeout 60s -short ./exporters/stdout/...
ok  	go.opentelemetry.io/otel/exporters/stdout	0.163s
go test -timeout 60s -short ./exporters/trace/jaeger/...
panic: close of closed channel

goroutine 83 [running]:
go.opentelemetry.io/otel/exporters/trace/jaeger.waitForCallWithTimeout.func1(0xc00030e4a0, 0x2, 0x2)
make: *** [test] Error 1
	/Users/runner/work/opentelemetry-go/opentelemetry-go/exporters/trace/jaeger/reconnecting_udp_client_test.go:100 +0x2a
github.com/stretchr/testify/mock.(*Mock).MethodCalled(0xc0003104b0, 0x17ba550, 0xe, 0xc00030e4a0, 0x2, 0x2, 0xc000310910, 0x5, 0x5)
	/Users/runner/go/pkg/mod/github.com/stretchr/testify@v1.7.0/mock/mock.go:452 +0xa1a
github.com/stretchr/testify/mock.(*Mock).Called(0xc0003104b0, 0xc00030e4a0, 0x2, 0x2, 0xc00030c898, 0x1, 0x0)
	/Users/runner/go/pkg/mod/github.com/stretchr/testify@v1.7.0/mock/mock.go:383 +0x1ab
go.opentelemetry.io/otel/exporters/trace/jaeger.(*mockResolver).ResolveUDPAddr(0xc0003104b0, 0x1504f32, 0x3, 0x15084df, 0xe, 0x1, 0x1462e00, 0x0)
	/Users/runner/work/opentelemetry-go/opentelemetry-go/exporters/trace/jaeger/reconnecting_udp_client_test.go:34 +0xec
go.opentelemetry.io/otel/exporters/trace/jaeger.(*reconnectingUDPConn).attemptResolveAndDial(0xc00030c6c0, 0xc000105f40, 0x2)
	/Users/runner/work/opentelemetry-go/opentelemetry-go/exporters/trace/jaeger/reconnecting_udp_client.go:90 +0x66
go.opentelemetry.io/otel/exporters/trace/jaeger.(*reconnectingUDPConn).reconnectLoop(0xc00030c6c0, 0x989680)
	/Users/runner/work/opentelemetry-go/opentelemetry-go/exporters/trace/jaeger/reconnecting_udp_client.go:82 +0x125
created by go.opentelemetry.io/otel/exporters/trace/jaeger.newReconnectingUDPConn
	/Users/runner/work/opentelemetry-go/opentelemetry-go/exporters/trace/jaeger/reconnecting_udp_client.go:62 +0x105
FAIL	go.opentelemetry.io/otel/exporters/trace/jaeger	0.372s
?   	go.opentelemetry.io/otel/exporters/trace/jaeger/internal/gen-go/agent	[no test files]
?   	go.opentelemetry.io/otel/exporters/trace/jaeger/internal/gen-go/agent/agent-remote	[no test files]
?   	go.opentelemetry.io/otel/exporters/trace/jaeger/internal/gen-go/jaeger	[no test files]
?   	go.opentelemetry.io/otel/exporters/trace/jaeger/internal/gen-go/jaeger/collector-remote	[no test files]
?   	go.opentelemetry.io/otel/exporters/trace/jaeger/internal/gen-go/zipkincore	[no test files]
?   	go.opentelemetry.io/otel/exporters/trace/jaeger/internal/gen-go/zipkincore/zipkin_collector-remote	[no test files]
FAIL
```

[full-log.txt](https://github.com/open-telemetry/opentelemetry-go/files/6407103/full-log.txt)

### Expected behavior

Panic to not happen.

### PR DESCRIPTION
**Pull request by zhyChesterCheung**
This PR adds logic to determine if the channel is closed to fix jaeger exporter test panic with close closed channel.

Resolves #1870

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -96,6 +96,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - BatchSpanProcessor now drops span batches that failed to be exported. (#1860)
 - Use `http://localhost:14268/api/traces` as default Jaeger collector endpoint instead of `http://localhost:14250`. (#1898)
 - Allow trailing and leading whitespace in the parsing of a `tracestate` header. (#1931)
+- Add logic to determine if the channel is closed to fix jaeger exporter test panic with close closed channel. (#1870)
```
Comment (MrAlias) [inline CHANGELOG.md:99]:
```suggestion
- Add logic to determine if the channel is closed to fix Jaeger exporter test panic with close closed channel. (#1870, #1973)
```

**Review by MrAlias — APPROVED** [review]

**Comment by zhyChesterCheung** [comment]
Hi, I'm a first-time contributor, could anybody help to approve running these workflows, great thx~

**Review by Aneurysm9 — APPROVED** [review]

**Comment by zhyChesterCheung** [comment]
When running the ci / test-coverage, something wrong happened with: 
{'detail': ErrorDetail(string='Unable to locate build via Github Actions API. Please upload with the Codecov repository upload token to resolve issue.', code='not_found')}
How to solve this problem, could we just rerun the workflow or anything else?

**Comment by MrAlias** [comment]
> When running the ci / test-coverage, something wrong happened with:
> {'detail': ErrorDetail(string='Unable to locate build via Github Actions API. Please upload with the Codecov repository upload token to resolve issue.', code='not_found')}
> How to solve this problem, could we just rerun the workflow or anything else?

It seems like the codecov endpoint is having issues today. I have seen quite a lot of this error in other PRs. Rerunning can resolve the issue as it looks like it is an instability on the receiving side, not our sending side.

**Comment by zhyChesterCheung** [comment]
Many thanks for helping solve the codecov problem, could you please help to merge the commit if it's convenient ? Thanks a lot ~

**Review by MadVikingGod — APPROVED** [review]

**Pull request merged**